Bottom: bae4ddc1e280471886b38e134fd6d42c57ada52a
Top:    c93e2238bbf040aee94664f94184a6137edded12
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-17 17:35:36 +0800

Refresh of virtio-net-correctly-transmit

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 59702f9..93ec469 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -788,7 +788,7 @@ static struct sk_buff *receive_mergeable(struct net_device *dev,
 			}
 			*xdp_xmit = true;
 			if (unlikely(xdp_page != page))
-				goto err_xdp;
+				put_page(page);
 			rcu_read_unlock();
 			goto xdp_xmit;
 		case XDP_REDIRECT:
